### ISSUE TITLE
Speed Ez - Account Ledger issues

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1411]
In GitLab by @kdjstudios on Mar 15, 2019, 14:55

**Submitted by:** <joann@speedez.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/7554503

**Approval List:** https://docs.google.com/spreadsheets/d/1KzxkNTsgff71b4Rau5Sw5y-P1-aJKv9fgz6_uYAuLYU/edit?usp=sharing

**Server:** External

**Client/Site:** Speed Ez

**Account:** Multiple

**Issue:** 

My feelings are that my accounts are never going to be correct.  Go to Account 5264, pull up the New Account Ledger and look at the Balance – then, print it out Pdf and look at the Balance – they are both different.  The Balance of $1437.08 is correct and NOT the Pdf printout of $977.48.  I know I am going to just keep finding these everywhere.  I look really stupid to my customers when I send them a report of a balance and then later have to explain to them that “Oh sorry, but the report was wrong”.  I also find it very hard to believe that this is just happening to me.  I need someone to go over every one of my accounts and find out what is wrong as I cannot deal with this much longer.  Thank you,

**List of all tickets with same issue of the account ledger:** #1389 #1423 #1420 #1407

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 15, 2019, 15:02

I wrote:

> Thank you for notifying us. We have informed the support team and will follow up with their research accordingly.
> 
> NOTE: This does appear to be the same concern that you reported in "Ledger Questions - 2019-02-26-73253"; which as we last followed up with you, that you had confirmed the support team had fixed the underlining issue on our development server. Please keep in mind that we have not had this fix released to production yet, as the support team is testing for any issues on production at the moment.  We will follow up on your first ticket when that fix is released to production and also follow up on this ticket if it is due to a difference issue. 

@smasih @amishra @pchaudhary @tim.traylor 

Please confirm that this is one of the accounts that the #1389 update to the ledger will fix. If not please explain the underlining issue and estimate a fix.

EDIT: If this is the same, please ensure we have the full list of accounts for internal and externals that list all that are effected by this change.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 18, 2019, 06:08

@kdjstudios let's discuss this tomorrow if Gary isn't on - or after his part if he is.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 19, 2019, 05:52

@kdjstudios @tim.traylor 

#1389 #1407 (tickets with account ledger issues)

Fix currently on dev

on ledger for first invoices we were pulling invoice finalized total instead of balance (new charges), which was right but this failed if payment on the first invoice showed up again because of sorting.

changed the way we calculated the balance on ledger.

seems like it's fixed on speedez except on account(where ledger seems correct) [balvslegspdz_old.csv](/uploads/7eeb43e253753748bcd09a4d5a3deabf/balvslegspdz_old.csv)

but can't be certain unless we ran the "Acc Bal vas Ledger Bal" scrip against both servers.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 19, 2019, 09:49

@smasih As discussed, please do run the comparison against all accounts and confirm that we have the accurate calculations now.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Mar 25, 2019, 05:53

Note: They're doing the comparisons and identifying code updates after each run to clear scenarios that are not matching.  As of today they're down to approx. 30 accounts that still have differences.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 25, 2019, 08:05

@kdjstudios 

Thank you for the update.

In above [comment] (http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_36982)
attached one invalid account **Sound Physicians** in the attachment. 
There are two invoices **SPDZ-19820** and **SPDZ-20253** which do have the calculation issues instead of the account ledger, which means by fixing those invoices only the account ledger will be correct.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Mar 28, 2019, 02:21

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 29, 2019, 24:24

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Mar 29, 2019, 24:30

@kdjstudios 

We would like to inform you that we have mentioned the list of all the ticket related to account ledger calculation on top of the description.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 1, 2019, 07:47

Hi [Kyle] ( @kdjstudios ) & [Tim] ( @tim.traylor ),

As suggested by Tim, we have run a comparison between the hosted production & newly developed code to find the number of discrepancies of account which are having ledger issues.

**For Hosted**

* On Production - 637 accounts having the ledger issues
* After New code change - Only 31 accounts having the ledger issues

Please have a look.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 1, 2019, 08:13

@pchaudhary 

Is there supposed to be a list of before and after; and a list of remaining issues along with what is causing those still? I am confused on what we are needing to look at.

Also I see this is only for Keener, correct? If so, then this looks like a huge issue as there are only 617 active accounts (1001 Total) for KNR. Along with KNRVA only having 188 active accounts (338 Total). That means that we have roughly 50% of total accounts have issues?

[KNRVA.xlsx](/uploads/4cf1b6a8f5dab56409bd1ca7d9c26920/KNRVA.xlsx)

[KNR.xlsx](/uploads/826914a7330afdbea7542a3fd64679fb/KNR.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 2, 2019, 06:42

Hi [Tim] ( @tim.traylor ) & [Kyle] ( @kdjstudios ),  

Sorry for the confusion. I think my last comment is misleading. I have updated that too & making it clear here.

We have run a script on both of the production servers to find the current number of accounts having the ledger issues & also run a script on a development server(with the latest DB) to find the number of remaining accounts having the ledger issues after we fixed the code.


| Server   | Total number of accounts | Existing number of faulty accounts | Faulty Active accounts | Number of accounts will remain (after code fix) |
|----------|--------------------------|------------------------------------|------------------------|-------------------------------------------------|
| Internal | 10054                    | 3610 [ledger_answernet.csv](/uploads/ffa58d23fce6d2d55f966cd84e088101/ledger_answernet.csv)                              | 1623                   | 1037  [ledger_answernet_code_fix.csv](/uploads/aef24eb109e6a80bc9c13428868fef91/ledger_answernet_code_fix.csv)                                          |
| Hosted   | 2598                     | 636  [ledger_hosted.csv](/uploads/07af4b516c67b359849e2d89dd98e8a1/ledger_hosted.csv)                              | 436                    | 31    [ledger_hosted_code_fix.csv](/uploads/d6ce8b972e03da4c7ad5e30b9b3e8bae/ledger_hosted_code_fix.csv)                                          |

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 2, 2019, 08:45

Marking for review with @tim.traylor

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 2, 2019, 09:21

@tim.traylor services/account_ledger.rb

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 2, 2019, 09:22

This was discussed today and @tim.traylor states we will need to have a column in the files above to show if an account is active/terminated. Currently we will most likely want to ignore terminated accounts.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 3, 2019, 05:00

@kdjstudios @tim.traylor 

We have updated the above files with the new column *Status* as True/False. Please check.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 3, 2019, 09:42

Marking this as the parent ticket now, since all development is being posted here.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 3, 2019, 10:30

@smasih @amishra @pchaudhary 

@tim.traylor informed me this was reviewed this morning and he did a deep dive with the team and the results did not appear to be correct. The team is going to recheck the accounts and post results of accounts that still have issues after the new code changes to better see the differences.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Apr 4, 2019, 03:49

Hi [Tim] ( @tim.traylor ), 

Please find the updated list of the accounts for both of the servers for prior & after code change.


| Server   | Total number of accounts | Existing number of faulty accounts | Faulty Active accounts | Number of accounts will remain (after code fix) |
|----------|--------------------------|------------------------------------|------------------------|-------------------------------------------------|
| Internal | 10054                    | 3610 [ledger_answernet.csv](/uploads/ffa58d23fce6d2d55f966cd84e088101/ledger_answernet.csv)                              | 1623                   | 1037  [ledger_answernet_code_fix.csv](/uploads/aef24eb109e6a80bc9c13428868fef91/ledger_answernet_code_fix.csv)                                          |
| Hosted   | 2598                     | 636  [ledger_hosted.csv](/uploads/07af4b516c67b359849e2d89dd98e8a1/ledger_hosted.csv)                              | 436                    | 31    [ledger_hosted_code_fix.csv](/uploads/d6ce8b972e03da4c7ad5e30b9b3e8bae/ledger_hosted_code_fix.csv)                                          |


**Note - The finalized total of the invoice is going to be the previous balance of the next invoice.**

*  There are some initial invoices of the account which do not have any line item but still they contain the finalized balance of the invoice. 

* Another example account which has the conversion date and is currently showing incorrect balance and with new code it's showing the right balance.   
Eg: **061-3302** ANKLE & FOOT SURGERY CENTER of Allentown


FYI current code for conversion date (in case we need to discuss)

````ruby
conv_date = account.site.site_conversion_date

      if conv_date.present? and conv_date != "" 
        invoices = Invoice.where(account_id: account._id, finalized: true,:created_at.gt => conv_date + 1.day)
        if invoices.count > 0
          payments = Payment.where(account_id: account._id, :created_at.gt => invoices.sort_by{|inv| inv.created_at}[0].created_at)
        else
          payments = Payment.where(account_id: account._id, :created_at.gt => conv_date + 1.day)
        end#invoices count check end
      else
        invoices = Invoice.where(account_id: account._id, finalized: true)
        payments = Payment.where(account_id: account._id)
      end
````

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 5, 2019, 05:43

@tim.traylor @rchauhan 

Ticket #877 introduced the concept of considering a conversion date on the ledger which was released to production. There was a number of fixes in place, but a chunk of the work was to be taken up by Thomas. I guess Thomas didn't complete it and the work item was lost in the transition. 

In January, this item was reported again, there was a hotfix implemented and the implementation was tested on the reported accounts. Both Kyle and us checked it and assumed that the fix was universal, without realizing that it was case specific. 

Once #1414(this ticket) was reported, I spotted that the fix was not working for all the accounts. After carefully checking the data I made some changes to the code and today after further changes the numbers have significantly improved, and maybe even permanently. 

Active accounts who's ledger and account balance don't match, number 20 on hosted and 40 on AnswerNet, and a first check on some of these seem to indicate that there conversion dates have not been set correctly. 

Richie asked me to put this as a hotfix on production as it is reporting and there is no data impact anywhere. 

A few other numbers that were discussed

After 877 Active Accounts that were not matching (Please note that approx 1000 new accounts have been added since this ticket)
* AnswerNet: `445`
* Hosted:    not sure

We restored the Jan 11 release and Active Accounts that were not matching 
* AnswerNet:  `1623`
* Hosted: `436`

After today's fix
* AnswerNet: `40` | [ledger_new_code_answernet.csv](/uploads/318a1ceb9ed7c4f8056920e5b10c97cc/ledger_new_code_answernet.csv)    
* Hosted: `20` | [ledger_new_code_hosted.csv](/uploads/fd4506ec43218f17a123018410c15324/ledger_new_code_hosted.csv)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Apr 5, 2019, 05:48

@kdjstudios 

original issue was raised by SpeedEz and their all accounts are fixed except `Sound Physicians	5381`. ( [+reason faulty invoice+] nothing to do do with the ledger calculation )

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Apr 5, 2019, 06:15

Now THAT is what I was expecting in differences!  40 out of 10,000+ makes sense and seems right.

Thank you for getting this squared away!

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 5, 2019, 08:22

@smasih @amishra @pchaudhary @tim.traylor 

Thank you for the update and releasing this directly to Production. 

I just want to confirm then the remaining accounts that have issues; most likely have issues with the conversion date as stated; their invoices are incorrect as also stated; or something other then the account ledger, which in turn is causing the account ledger to show incorrectly. If so, what is needing to be done to fix the remaining accounts? Please provide an estimate. 

Marking back for development, as we do not want to notify any of operations and/or clients until all accounts are correct.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 18, 2019, 06:03

assigned to @pchaudhary and unassigned @amishra

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 18, 2019, 08:25

We will need a game plan for resolving the roughly 50 accounts. This could be due to multiple issues and require a lot of research and development. We will want to review with Gary for prioritization and expectations.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 25, 2019, 08:35

@amishra @smasih @pchaudhary 

Please look into three of the real accounts and find what the cause to the discrepancy is. Hopefully it is a common and easy thing to spot and each account doesn't have a different reason. We will want to be able to review this with Gary.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2019, 09:38

@amishra @smasih @pchaudhary 

This was reviewed today with Gary. 

Most recent stuff at the top, older at the bottom (#1374)

AR first, then fix these next, we need to have the Ledger fixed and working. 
- Can we give the sites a method to fix these on their own?
 - This way less programming time and addressed below:


First, get to Gary the list of sites and accounts that this is occurring at. 
- Create a code for nothing but adjusting the report (does not update anything but the report)
- Ops will then adjust the accounts.
- We will need a document that has detailed instructions on how to do this so then ops does not need to contact us.
 - We will want this to be done the week following the closing of the billing.
 - Only shows up on the ledger, does not show up on the account balance. Note will need verbiage "Account ledger was out of balance for unknown reason, this was placed here to correct this."

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2019, 10:58

@amishra @smasih @pchaudhary @tim.traylor 

Any updates? We now have additional tickets being submitted about the accounts that are still in error. This needs to be resolved.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on May 14, 2019, 03:40

@kdjstudios @tim.traylor 

> Can we give the sites a method to fix these on their own?
>
> - This way less programming time and addressed below:

* Agreed. We can make a fee type which can be used by Ops to adjust the Account Ledger balance to the Account Balance.

> First, get to Gary the list of sites and accounts that this is occurring at.

* It is already there in the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_37872


> Create a code for nothing but adjusting the report (does not update anything but the report)

* As far as I can understand, that this should be a fee (staged/special account ledger) fee which will be used to adjust the account ledger of faulty accounts.

Rough Estimate - 

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|Medium-1|4|Concepts and Implementation Well Understood|5|Low Impact|1.5|6.5|

* We can provide a better close estimate once we have a mockup.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 7, 2019, 10:20

Approved.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 12, 2019, 06:30

@kdjstudios 

It would be great if you can provide the Mockup of the Fee Adjustment functionality or we have to create it on our own while the preferred development policies. Please let us know.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 12, 2019, 09:33

@pchaudhary @tim.traylor @smasih @amishra 

I am confused on how there was an estimate created if a mock up was needed? Can you please explain what you are needing mocked up in detail that was not apart of your estimate. Is this not simply creating a Billing Code that only effects the account ledger?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 13, 2019, 02:01

@kdjstudios 

> I am confused on how there was an estimate created if a mock-up was needed? Can you please explain what you are needing to be mocked up in detail that was not apart of your estimate. Is this not simply creating a Billing Code that only effects the account ledger?

* The estimation is the process estimation as the only display code will not take that much time. 

* We need to discuss it further with @tim.traylor. We need the clarification on the Billing Code process. 

* Do we need to simply add a Code which would affect only the Account Ledger?

* From where do we need to add that code? - **Account Ledger Page / Account Edit Page / Both**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 13, 2019, 09:40

@pchaudhary 
1. This will require a new collection to track the adjustment amount and the date the adjustment was created.
1. Tim requested that the adjustment mechanism only be accessible to O+1 and above (o6 for Answernet) or 'manager' level (for externals) security levels.  
1. The adjustment button should be on the account ledger screen and only visible if the ledger does not match the account balance.
1. When clicked, the button should display the amount that it is going to apply to bring the ledger in balance with the account balance.  They can then click 'Ok' to create the entry in the new collection.
1. The ledger will be modified to also query this new collection and bring in any adjustment entries in the correct order based on the date.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 13, 2019, 09:41

Verbiage for the note/line account ledger: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_39288

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 17, 2019, 05:45

@kdjstudios @tim.traylor 

> Tim requested that the adjustment mechanism only be accessible to O+1 and above (o6 for Answernet) or 'manager' level (for externals) security levels

* Internal - O2, O3, O4, O5 & O6 for Internal.
* External - manager

* Do we need to add `managerwtr` for Externals.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 17, 2019, 05:55

@pchaudhary Pls do include `managerwtr` in the levels with permission to do this.
Only o5 and o6 for internal.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 17, 2019, 23:31

@kdjstudios @tim.traylor 

Thanks for the update. We have successfully developed the functionality.

We also would like to confirm that should we add **Admin** also can access the functionality?


* We request Kyle to test the functionality on Merge2 Server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 17, 2019, 23:58

changed time estimate to 6h 30m

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 18, 2019, 05:32

@pchaudhary Hi Piyush, Admin is God in our systems - they are the highest level of permission.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 18, 2019, 08:47

@pchaudhary @tim.traylor 

I was able to run through a few test accounts both on the list and not on the list. The adjust button only showed for accounts that were mismatched. It automatically calculated the correct amount and allowed for the user to confirm or cancel accordingly. 

I did run into one concern, which I took a screen shot below. It would appear we are having a decimal issue? I expected to see it calculate $0.01; but what is is presenting alarms me to think what is going on. 

![Adjust](/uploads/a19246fe05cb4f9694d74a20a2c1d857/Adjust.png)

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 18, 2019, 08:57

@amishra @pchaudhary 

I know every thing else is intact and working **but I don't want these decimal issues to be raised again..!! please follow the existing convention unless told otherwise**

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 19, 2019, 02:20

@kdjstudios @smasih 

We have fixed the round off issue on for this functionality on the Merge2 server. Please test.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 19, 2019, 07:55

@pchaudhary 

Thank you, I tested this and am marking it for announcement.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 22, 2019, 03:48

@kdjstudios 


We have successfully released on both of the production servers.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 24, 2019, 10:50

@amishra 

Thank you. May we please have an updated list of accounts that still are not matching. The last list was from about 2 months ago and might be outdated now http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_37872

Once we have the latest list of accounts, I will send out notices to all sites/clients and follow up with them until there are no remaining accounts.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2019, 12:08

Moving back to development so we can have the updated list of accounts.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 26, 2019, 04:33

@kdjstudios @tim.traylor 

Please find the list of the accounts having the wrong ledger.

* **Internal**
37 Active Accounts - [latest_account_ledger_internal.csv](/uploads/7b9dc76bb8aaa203f249a0f337bbca47/latest_account_ledger_internal.csv)

* **Hosted**
20 Active Accounts - [latest_account_ledger_hosted.csv](/uploads/8f49e6283df308af8e1b3a9ef41e77e8/latest_account_ledger_hosted.csv)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 26, 2019, 07:51

removed milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2019, 10:29

I am writing emails to all of our clients and sites accordingly. I am only including the active accounts for their manual fix. The terminated ones, we do not care about correct @tim.traylor ?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 26, 2019, 14:41

@kdjstudios I would think they only care about active accounts, but we could probably give them the terms as well just so everything is covered.  Did we ever get a count/list that has the termed accounts as well?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2019, 16:11

@tim.traylor 

The above spreadsheet actually has the terms on it too, I was just filtering it down. There is quite a larger amount with the terms included. I wonder if doing this programmaticly rather then having ops and clients do this would be better. We would of course need to send out a notice prior to and after with the accounts that would be adjusted. 

Internal - 250 accounts total

External - 42 accounts total

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jun 27, 2019, 05:20

@kdjstudios If it's 250 for internal then I agree that would be a better option. @pchaudhary Piyush, how long would it take to write a script to automatically create the new transactions for all of these accounts?

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jun 27, 2019, 05:59

@tim.traylor 

Yes, we agree. We can run a script for the Terminated accounts on both of the Servers or you would like us to run it for all of the accounts either Active or Terminated.

> Piyush, how long would it take to write a script to automatically create the new transactions for all of these accounts?

|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|2.5|Low Impact|0.5|3|

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 28, 2019, 12:49

@smasih @amishra @pchaudhary @tim.traylor 

Was this released to production without approval? We have not even been able to send out notices to operations or users? #1516

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 30, 2019, 22:46

@kdjstudios 

Thank you for the update.

I would like to inform you that you had tested and approved for ready.   
As per your comment http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_42512 you marked it for 
the announcement then ready.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2019, 08:54

@amishra @smasih 

NO; There is no approval for the estimate http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_43099

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Jul 1, 2019, 23:21

@kdjstudios 

We did not run the script for all of the accounts as the estimate is not approved yet. Ticket #1516 is the adjustment made by the User.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 2, 2019, 11:03

@pchaudhary 

Thank you for confirming.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 5, 2019, 10:23

@tim.traylor 

Below is the draft email to send out. Please let me know your feedback and then I may begin sending it out. Also do we want to review this with Gary?

> Dear SA Billing Family Member,
> 
> We have received reports of account ledgers not matching the account balance. After in-depth research by our support team they were able to determine that there are historical invoicing issues. The account's balance is correct; but since the account ledger is using the historically invoicing for calculating its totals, which has discrepancies, it is resulting the the account ledger not matching. After review and discussion, due to the amount of invoices that teams and operations would need to check on each account in order to find the invoice(s) that have an issue we found it would most easily be resolved with a simple one time calculation adjustment to the account ledger. 
> 
> We have attached a list of all accounts that currently do not match between the account ledger and the account balance. With your review and approval the support team will automatically apply the adjust to these accounts. 
> 
> If you would like to apply the adjustment your self, you may do so by navigating to the account ledger page and near the bottom clicking the "Adjust Ledger" button. It will automatically calculate the adjustment amount and present you with an confirmation box. Upon clicking confirm an adjustment line will be added to the account ledger and all forward going calculations will be fixed. 
> 
> Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Jul 8, 2019, 05:43

Dear SA Billing Family Member,

We have received reports of a small number of account ledgers not matching the account balance. After in-depth research by our support team they were able to determine that these are due to historical invoicing issues. The account's balance is correct, but since the account ledger is using historical invoicing for calculating its totals, it is resulting the the account ledger not matching. 

We have created a mechanism by which SAB users can adjust the account ledger, and we are planning to automatically apply the adjustments to the account ledger.  We have attached a list of all accounts that currently do not match between the account ledger and the account balance. With your review and approval the support team will automatically apply the adjust to these accounts.

If you would like to apply the adjustment your self, you may do so by navigating to the account ledger page and near the bottom clicking the "Adjust Ledger" button. It will automatically calculate the adjustment amount and present you with an confirmation box. Upon clicking confirm an adjustment line will be added to the account ledger and all forward going calculations will be fixed.

Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 10, 2019, 11:56

I have just finished separating out each site/clients account into their own list and have begun sending out the emails. Below is the link to monitor the approvals and feedback for each site/client.

https://docs.google.com/spreadsheets/d/1KzxkNTsgff71b4Rau5Sw5y-P1-aJKv9fgz6_uYAuLYU/edit?usp=sharing

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 10, 2019, 11:56

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Aug 30, 2019, 06:34

@kdjstudios Hi Kyle, is there an update on this?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 30, 2019, 14:23

@tim.traylor 

I am still waiting on quite a few sites for their approvals. I will follow up with them next week on Tuesday time permitting.

I think on this we are just needing your approval for the estimate on http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1414#note_43099. once we have that and the sites approvals then the team will be able to run the updates.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 3, 2019, 13:23

I was able to follow up on the emails. We have approval to make the changes for the sites that have not already done this themselves.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 4, 2019, 02:13

@kdjstudios 

Do we need to run the script on our regular release or we have to do a special release for that?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 4, 2019, 09:05

@pchaudhary @tim.traylor 

Lets discuss this on tomorrows meeting. I would think a regular release would be acceptable since this has been a known and ongoing issue for 5 months now.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Sep 14, 2019, 05:13

@kdjstudios 

We ran successfully the script on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 1, 2019, 10:54

closed
